src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -100,20 +102,21 @@ public Task StartReceive(CancellationToken cancellationToken)
 
             for (var i = 0; i < numberOfPumps; i++)
             {
-                pumpTasks.Add(Task.Run(() => ConsumeMessages(tokenSource.Token), CancellationToken.None));
+                pumpTasks.Add(Task.Run(() => ConsumeMessages(messagePumpCancellationTokenSource.Token), cancellationToken));
             }
 
             return Task.CompletedTask;
         }
 
-        public async Task StopReceive(CancellationToken cancellationToken)
+        public async Task StopReceive(CancellationToken cancellationToken = default)
         {
-            if (tokenSource == null)
+            if (messagePumpCancellationTokenSource == null)
```

Comment:
This is the only transport that guards for multiple calls to stop, should we align the others?

Reply:
Core already makes stopping a stopped endpoint a no-op: https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs#L25-L28

If we're going to spend cycles aligning anything, I'd like it to be in removing the duplicate code in the first place.